[Improvement-13824][Resource Center] User can only visit resources under tenant defaultPath

Radeity opened the pull request:
<!--Thanks very much for contributing to Apache DolphinScheduler. Please review https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html before opening a pull request.-->

## Purpose of the pull request
- This PR closes https://github.com/apache/dolphinscheduler/issues/13824
<!--(For example: This pull request adds checkstyle plugin).-->

## Brief change log
- Add new api `/resource/base-dir` for querying base resource dir which add constraint that breadcrumb navigation can only visit resources or files under tenant's resource paths.
<!--*(for example:)*
- *Add maven-checkstyle-plugin to root pom.xml*
-->

## Verify this pull request
- Add some UT.

songjianet commented on dolphinscheduler-ui/src/views/resource/index.tsx:
```diff
@@ -25,4 +27,15 @@ const resource = defineComponent({
 })
 
+export function getBaseDir(resourceType: ResourceType) {
+  const baseDir = ref(String(""));
```
Please use single quotes and remove the semicolons.

songjianet commented on dolphinscheduler-ui/src/views/resource/index.tsx:
```diff
@@ -15,7 +15,9 @@
 
-import { defineComponent } from 'vue'
+import { defineComponent, ref } from 'vue'
+import { queryBaseDir } from "@/service/modules/resources";
```
Please remove the semicolon.

songjianet commented on dolphinscheduler-ui/src/views/resource/index.tsx:
```diff
@@ -15,7 +15,9 @@
-import { defineComponent } from 'vue'
+import { defineComponent, ref } from 'vue'
+import { queryBaseDir } from "@/service/modules/resources";
+import { ResourceType } from "@/views/resource/components/resource/types";
```
Please remove the semicolon.

songjianet commented on dolphinscheduler-ui/src/views/resource/components/resource/index.tsx:
```diff
@@ -46,6 +46,7 @@ import styles from './index.module.scss'
 import type { Router } from 'vue-router'
 import Search from "@/components/input-search"
 import { ResourceType } from "@/views/resource/components/resource/types";
```
Please remove the semicolon.

songjianet commented on dolphinscheduler-ui/src/views/resource/components/resource/index.tsx:
```diff
@@ -46,6 +46,7 @@ import styles from './index.module.scss'
 import type { Router } from 'vue-router'
 import Search from "@/components/input-search"
 import { ResourceType } from "@/views/resource/components/resource/types";
+import { getBaseDir } from "@/views/resource";
```
Please remove the semicolon.

songjianet commented on dolphinscheduler-ui/src/service/modules/resources/index.ts:
```diff
@@ -45,6 +45,16 @@ export function queryResourceListPaging(
 }
 
+export function queryBaseDir(
+    params: ResourceTypeReq,
```
Please remove the comma.

Radeity committed:
[Improvement-13824][Resource Center] User can only visit resources under tenant defaultPath

Radeity committed:
remove semicolon

Radeity committed:
set null in some specific scenario

Radeity commented:
Hi, @songjianet , thanks for your comments, i've removed all semi-colons and modified the logic that only query `base-dir` after login and save it in `userStore`.

songjianet commented:
> Hi, @songjianet , thanks for your comments, i've removed all semi-colons and modified the logic that only query `base-dir` after login and save it in `userStore`.

![image](https://user-images.githubusercontent.com/19239641/231921513-6e3f5e42-a3f9-4b63-995a-2e8672409927.png)

Radeity committed:
remove useless console.log

Radeity commented:
> > Hi, @songjianet , thanks for your comments, i've removed all semi-colons and modified the logic that only query `base-dir` after login and save it in `userStore`.
> 
> ![image](https://user-images.githubusercontent.com/19239641/231921513-6e3f5e42-a3f9-4b63-995a-2e8672409927.png)

Done.

Radeity commented:
Hi, @SbloodyS @caishunfeng , would you like to help review back-end part when you're available :D

SbloodyS reviewed (approved):
+1

caishunfeng commented:
This pr fix the security problem, and I think we should cherry-pick to the next 3.1.x version, so I add the `3.1.x` label @zhongjiajie

zhongjiajie commented:
> This pr fix the security problem, and I think we should cherry-pick to the next 3.1.x version, so I add the `3.1.x` label @zhongjiajie

cc @zhuangchong

zhuangchong commented:
@Radeity The part code in the dev branch and the 3.1.9 branch has been greatly changed in the resource center. Can you submit a PR to the 3.1.9-prepare branch?

zhongjiajie commented:
> @Radeity The part code in the dev branch and the 3.1.9 branch has been greatly changed in the resource center. Can you submit a PR to the 3.1.9-prepare branch?

Or maybe we can tell to security or reporter we could fix it in 3.2.1? @caishunfeng

Radeity commented:
> @Radeity The part code in the dev branch and the 3.1.9 branch has been greatly changed in the resource center. Can you submit a PR to the 3.1.9-prepare branch?

Sorry for the late reply, currently, I'm busy with my thesis, do not have much time on this, can we consider @z
's opinion?
